internal/reconciler/tracepipeline/reconciler.go

```diff
@@ -286,3 +297,72 @@ func (r *Reconciler) getReplicaCountFromTelemetry(ctx context.Context) int32 {
 	}
 	return defaultReplicaCount
 }
+
+func (r *Reconciler) cleanUpOldTraceCollectorResources(ctx context.Context) error {
+	oldTraceCollectorResources := []client.Object{
+		&corev1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "telemetry-trace-collector",
+				Namespace: "kyma-system",
+			},
+		},
+		&rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "telemetry-trace-collector",
+			},
+		},
+		&rbacv1.ClusterRole{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "telemetry-trace-collector",
+			},
+		},
+		&corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "telemetry-trace-collector-metrics",
+				Namespace: "kyma-system",
+			},
+		},
+		&networkingv1.NetworkPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "telemetry-trace-collector",
+				Namespace: "kyma-system",
+			},
+		},
+		&corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "telemetry-trace-collector",
+				Namespace: "kyma-system",
+			},
+		},
+		&corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "telemetry-trace-collector",
+				Namespace: "kyma-system",
+			},
+		},
+		&appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "telemetry-trace-collector",
+				Namespace: "kyma-system",
+			},
+		},
+	}
+
+	if r.istioStatusChecker.IsIstioActive(ctx) {
+		oldTraceCollectorResources = append(oldTraceCollectorResources,
+			&istiosecurityclientv1.PeerAuthentication{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "telemetry-trace-collector",
+					Namespace: "kyma-system",
+				},
+			})
+	}
+
+	for _, oldResource := range oldTraceCollectorResources {
+		if err := r.Delete(ctx, oldResource); err != nil && !apierrors.IsNotFound(err) {
+			return fmt.Errorf("failed to delete one of the old trace collector resources: %w", err)
+		}
+	}
```

Comment:
I suggest modifying the error handling logic to attempt deleting all possible resources first. If any deletion fails, an error can be returned at the end. You can use https://pkg.go.dev/errors#Join for that.

Reply:
Alright, thank you!
I will do the changes tomorrow morning